misc/init.d/cfengine3.in

```diff
@@ -426,9 +426,13 @@ cf_status_daemon()
         echo "Warning: PID file '$pidfile' does not contain the right PID ('$pids' != '$daemon_status'). Should restart CFEngine."
     fi
 
-    # Lack of quotes around $pids is important to eliminate newlines.
-    if [ -n "$(echo $pids | grep '[^0-9]')" ]; then
-        echo "Warning: Multiple $base_daemon processes present. Should restart CFEngine."
+    # cf-consumer is expected to have multiple pids, so if checking status for
+    # cf-consumer, skip detection of multiple pids to avoid false alarms
+    if [ ! "${base_daemon}" = "cf-consumer" ]; then
```

Comment:
Use `!=` operator instead of `!` (not) operator. The latter is not supported by some shells.